docs/uppy-core.mdx

```diff
@@ -0,0 +1,1413 @@
+
+This ID should be persistent across page reloads and navigation—it shouldn’t be a random number
+that is different every time Uppy is loaded.
+:::
+
+#### `autoProceed`
+
+Upload as soon as files are added (`boolean`, default: `false`).
+
+By default Uppy will wait for an upload button to be pressed in the UI,
+or the `.upload()` method to be called before starting an upload.
+Setting this to `true` will start uploading automatically after the first file is selected
+
+#### `allowMultipleUploadBatches`
+
+Whether to allow several upload batches (`boolean`, default: `true`).
+
+This means several calls to `.upload()`,
+or a user adding more files after already uploading some.
+An upload batch is made up of the files that were added since the earlier `.upload()` call.
+
+With this option set to `true`, users can upload some files, and then add _more_ files and upload those as well.
+A model use case for this is uploading images to a gallery or adding attachments to an email.
+
+With this option set to `false`, users can upload some files, and you can listen for the [`'complete'`](/docs/uppy/#complete)
+event to continue to the next step in your app’s upload flow.
+A typical use case for this is uploading a new profile picture.
+If you are integrating with an existing HTML form, this option gives the closest behaviour to a bare `<input type="file">`.
+
+#### `debug`
+
+Whether to send debugging and warning logs (`Boolean`, default: `false`). 
+
+Setting this to `true` sets the [`logger`](#logger) to [`debugLogger`](#debuglogger).
+
+#### `logger`
+
+Logger used for [`uppy.log`](#uppy-log) (`Object`, default: `justErrorsLogger`).
+
+By providing your own `logger`, you can send the debug information to a server, choose to log errors only, etc.
+
+:::note
+Set `logger` to `Uppy.debugLogger` to get debug info output to the browser console:
+
+```js
+import Uppy from '@uppy/core'
+
+const uppy = new Uppy({
+  logger: Uppy.debugLogger,
+})
+```
+:::
+
+:::note
+You can also provide your own logger object:
+it should expose `debug`, `warn` and `error` methods, as shown in the examples below.
+
+Here’s an example of a `logger` that does nothing:
+
+```js
+const nullLogger = {
+  debug: (...args) => {},
+  warn: (...args) => {},
+  error: (...args) => {},
+}
+```
+:::
+
+#### `restrictions`
+
+Conditions for restricting an upload (`Object`, default: `{}`).
+
+| Property             | Value             | Description                                                                                                                      |
+| -------------------- | ----------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| `maxFileSize`        | `null`, `Number`  | maximum file size in bytes for each individual file                                                                              |
```

Comment:
do we need to specify null as a value? Then shouldn't `undefined` also be accepted? can't we instead say `Number?` indicating that it's optional?